torch/csrc/aot_inductor/c/aten_shim.h

```diff
@@ -0,0 +1,116 @@
+#pragma once
+
+#include <stdint.h>
+#include <torch/csrc/aot_inductor/c/utils.h>
+
+// What problems are we trying to solve here with a C shim layer?
+//
+// If we Ahead-of-Time compile a model using AOTInductor and directly call aten
+// ops or use aten/c10 data structures in the generated code, we will end up
+// with ABI compatibility issues. People working on aten/c10 C++ code may not
+// know there are some AOTInductor compiled model code relying on aten/c10 code
+// to be ABI stable. By introducing a C shim layer here, we are minimizing the
+// surface that could cause ABI breakage. AOTInductor generated code will
+// NOT be allowed to call any of aten/c10 C++ APIs. It is only allowed to call
+// functions declared in this header file instead.
+
+// WARNING: Change the following types can break ABI compatibility
+// Typedef these to be exactly the same as c10 since C enum types can not be
+// specified as int8_t. We will assert the data type length at run time.
+typedef int8_t AOTInductorDeviceType;
+typedef int8_t AOTInductorDeviceIndex;
+typedef int8_t AOTInductorScalarType;
```

Comment:
To be honest, it's still not clear at all to me why we need to pass device type/index / scalar type around. In particular, we're going to have a thrift sidecar for the FFI calls right? So we can also use that to do argument type checking entirely on the runtime side.

Reply:
Also, IMO, there isn't really an reason to use a narrower type here. You're only passing these in as arguments, but that doesn't change the ABI at all https://godbolt.org/z/TG44PrWjj

And the downside to def'ing them too narrow is that if you ever do need to expand it, it's very painful.